### ISSUE TITLE
Wishart samples are always Float64, regardless partype

### ISSUE THREAD

**Issue by johnczito** [JuliaStats/Distributions.jl#960]
To clarify, this is not the expected behavior, correct?

```julia 
julia> using Distributions, LinearAlgebra

julia> d = Wishart(Float32(4), Matrix{Float32}(I, 2, 2))

julia> partype(d)
Float32

julia> rand(d)
2×2 Array{Float64,2}:
 1.47454  1.53805
 1.53805  6.80249
```

The sample should be `Array{Float32,2}`. If so, I'll submit a PR to fix.

**Comment by mschauer** [comment]
In the fix, also add `eltype` ... or wait for #951

**Comment by richardreeve** [comment]
Yes, it turns out that almost every single distribution fails to return the right type from quantile() and most from rand(). I'm going through fixing them all as part of #951.

**Comment by johnczito** [comment]
@richardreeve Very good. Thanks for that. Most of the matrix-variates rely on the Wishart for sampling at some point. So fixing it will be a big step toward straightening all of them out.
